app/views/output_elements/index.html.haml

```diff
@@ -1,10 +1,10 @@
 #select_charts
   - @groups.each_pair do |group, charts|
-    - if charts.any?
-      %h1= t("chart_group.#{group.downcase}")
-      - charts.each do |chart|
-        %div.pick_chart{'data-chart_id' => chart.id, 'data-chart_holder' => @chart_holder}
-          %a
-            = image_tag "output_elements/#{chart.icon}", :size => "50x30"
-            %h2= t("output_elements.#{chart.key}").html_safe
+    %h1= t("chart_group.#{group.downcase}")
+    - if charts.is_a?(Array)
+      = render 'chart_preview', charts: charts
 
+    - elsif charts.is_a?(Hash)
+      - charts.each_pair do |sub_group, charts|
+        %h2= sub_group ? t("chart_sub_group.#{sub_group.downcase}") : 'Ungrouped'
```

Comment:
Tiny thing: Could we do away with the "Ungrouped" header? Just show the ungrouped charts at the top of the section without that `<h2>`? I don't think a visitor will care that those charts are ungrouped, so that header doesn't have much meaning to them.